### PR TITLE
Add JWT authentication with refresh tokens

### PR DESCRIPTION
**Pull request by SpaskoKatsarski**
Added `SignIn`, `SignUp`, `SignOut` functionality. Generating `access` and `refresh` tokens upon successful login which contain `userId`, `email` and `username` as claims. Extending the `IdentityUser`.

### CODE REVIEW

File: WarehouseManagement.Api/Controllers/DifferenceTypeController.cs [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
 using System.Security.Claims;
 using WarehouseManagement.Core.Contracts;
 using WarehouseManagement.Core.DTOs.DifferenceType;
 using static WarehouseManagement.Common.MessageConstants.Keys.DifferenceTypeMessageKeys;
 
 namespace WarehouseManagement.Api.Controllers;
 
+[Authorize]
```
Comment (vexleet) [inline WarehouseManagement.Api/Controllers/DifferenceTypeController.cs:10]: dont we need this for all controllers?

File: WarehouseManagement.Api/Controllers/DeliveryController.cs [hunk 2]
```diff
@@ -41,6 +42,7 @@ public async Task<IActionResult> GetDelivery(int id)
 
     [HttpGet("all")]
     [ProducesResponseType(200, Type = typeof(PageDto<DeliveryDto>))]
+    [ProducesResponseType(401)]
```
Comment (vexleet) [inline WarehouseManagement.Api/Controllers/DeliveryController.cs:45]: same for this one?

File: WarehouseManagement.Api/Program.cs [hunk 3]
```diff
@@ -22,6 +23,24 @@
...
 builder.Services.AddApplicationService();
 
+builder.Services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+ .AddJwtBearer(options =>
+ {
+     options.TokenValidationParameters = new TokenValidationParameters
+     {
+         ValidIssuer = builder.Configuration["Jwt:Issuer"],
+         ValidAudience = builder.Configuration["Jwt:Audience"],
+         IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(builder.Configuration["Jwt:Key"]!)),
```
Comment (vexleet) [inline WarehouseManagement.Api/Program.cs:33]: i guess we need values for these inside the appsettings?

File: WarehouseManagement.Api/Controllers/DifferenceTypeController.cs (hunk 1 above)
Comment (SpaskoKatsarski) [reply WarehouseManagement.Api/Controllers/DifferenceTypeController.cs:10]: Yes I put it just to test it. We should discuss which routes will be accessible only by authorized users.

File: WarehouseManagement.Api/Program.cs (hunk 3 above)
Comment (SpaskoKatsarski) [reply WarehouseManagement.Api/Program.cs:33]: We currently have:
image

**Pull request merged**